src/alt.js

```diff
@@ -219,18 +219,30 @@ const StoreMixin = {
 const setAppState = (instance, data, onStore) => {
   const obj = JSON.parse(data)
   Object.keys(obj).forEach((key) => {
-    assign(instance.stores[key][STATE_CONTAINER], obj[key])
-    onStore(instance.stores[key])
+    const store = instance.stores[key]
+    if (store[LIFECYCLE].deserialize) {
+      obj[key] = store[LIFECYCLE].deserialize(obj[key])
+      /* istanbul ignore next */
+      if(!obj[key]) {
+        throw new Error(
+          `The "deserialize" lifecycle method in the store, ${key}, must return data.`
+        )
+      }
+    }
+    assign(store[STATE_CONTAINER], obj[key])
+    onStore(store)
   })
 }
 
 const snapshot = (instance) => {
   return JSON.stringify(
     Object.keys(instance.stores).reduce((obj, key) => {
-      if (instance.stores[key][LIFECYCLE].snapshot) {
-        instance.stores[key][LIFECYCLE].snapshot()
+      const store = instance.stores[key]
+      let customSnapshot
+      if (store[LIFECYCLE].snapshot) {
+        customSnapshot = store[LIFECYCLE].snapshot()
       }
-      obj[key] = instance.stores[key].getState()
+      obj[key] = customSnapshot ? customSnapshot : store.getState()
```

Comment:
Didn't make this change. Because then it makes `snapshot` expect a return value.  Keeping it the way it is, is a bit more verbose, but always uses the default unless there is a valid return value.

I didn't catch it, but the tests did lol